core/kernel/interrupt.c

```diff
@@ -76,12 +76,32 @@ void itr_add(struct itr_handler *h)
 	SLIST_INSERT_HEAD(&handlers, h, link);
 }
 
-void itr_enable(struct itr_handler *h)
+void itr_enable(size_t it)
 {
-	itr_chip->ops->enable(itr_chip, h->it);
+	itr_chip->ops->enable(itr_chip, it);
 }
 
-void itr_disable(struct itr_handler *h)
+void itr_disable(size_t it)
 {
-	itr_chip->ops->disable(itr_chip, h->it);
+	itr_chip->ops->disable(itr_chip, it);
+}
+
+int itr_raise_spi(size_t it)
```

Comment:
This function and `itr_raise_ppi()` are identical, wouldn't one `itr_raise_pi()` be enough?